package-manifest-spec.md

```diff
@@ -0,0 +1,156 @@
+# Package Manifest Specification
+
+This document defines the specification for the **Package Manifest**.  The
+package manifest contains meta data about the package as well as dependencies
+needed for installation of the package.
+
+
+## Guiding Principles
+
+The package manifest specification makes the following assumptions about the
+document lifecycle.
+
+1. Package manifests are long lived documents that will be modified regularly
+2. Package manifests will primarily be consumed by package managers.
+3. Package manifests will primarily be used by package indexes to popululate basic package metadata.
+2. Package Managers will typically be used to assist in the creation and modification of Package Manifest documents.
+4. Package Manifest documents will occasionally be created and modified manually by people.
+4. Package manifests will typically be stored alongised the package source code.
+
+
+## Format
+
+The canoonical format for the package manifest is a JSON document containing a
+single JSON object.  
+
+
+## Filename
+
+Convention is to name this document `epm.json` which is short for *'ethereum
+package manifest'*.
+
+
+## Document Specification
+
+The following fields are defined for the package manifest.  Custom fields may
+be included.  Custom fields **should** be prefixed with `x-` to prevent name
+collisions with future versions of the specification.
+
+
+### Manifest Version: `manifest_version`
+
+
+The `manifest_version` field defines the version of the ethereum package manifest
+specification (this document) that this manifest conforms to. All manifests
+**must** include this field.
+
+* Key: `manifest_version`
+* Type: Integer
+* Allowed Values: `1`
+* Package Manager Guide: Package managers should validate this field is valid prior to publishing new manifests.
+
+
+### Package Name: `package_name`
+
+The `package_name` field defines a human readable name for this package.  All
+manifests **should** include this field.
+
+* TODO: what should happen if this is missing (should it be required?)
+* Key: `package_name`
+* Type: String
+
+### Authors: `authors`
+
+The `authors` field defines a list of human readable names for the authors of
+this package.  All manifests **should** include this field. 
+
+
+* Key: `authors`
+* Type: List of Strings
+
+
+### Version: `version`
+
+The `version` field declares the current version number of this package.  This value
+**should** be conform to the [semver](http://semver.org/) version numbering
```

Comment:
If people want to use whole number versioning I don't want to stop them.  I agree that it's good for people to use this standard but there are lots of legitimate projects that don't use semver and I'm hesitant to put this as a requirement.